app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -183,13 +192,24 @@ abstract class TwoPaneFragment : Fragment() {
     }
 
     fun openThread(uid: String) {
+
         mainViewModel.currentThreadUid.value = uid
-        slidingPaneLayout.open()
+
+        val isOpening = slidingPaneLayout.openPane()
+
+        if (isOpening) requireActivity().window.statusBarColor = requireContext().getColor(R.color.backgroundColor)
     }
 
     private fun resetPanes() {
-        slidingPaneLayout.close()
+
+        val isClosing = slidingPaneLayout.closePane()
+
+        if (isClosing && this is ThreadListFragment) {
+            requireActivity().window.statusBarColor = requireContext().getColor(R.color.backgroundHeaderColor)
+        }
```

Comment:
Shouldn't this be called when the `currentThreadUid` is observed as well?

Reply:
eDone in https://github.com/Infomaniak/android-kMail/pull/1606